checkstyle/suppressions.xml

```diff
@@ -92,7 +92,7 @@
               files="(Utils|Topic|KafkaLZ4BlockOutputStream|AclData|JoinGroupRequest).java"/>
 
     <suppress checks="CyclomaticComplexity"
-              files="(AbstractFetch|ConsumerCoordinator|CommitRequestManager|FetchCollector|OffsetFetcherUtils|KafkaProducer|Sender|ConfigDef|KerberosLogin|AbstractRequest|AbstractResponse|Selector|SslFactory|SslTransportLayer|SaslClientAuthenticator|SaslClientCallbackHandler|SaslServerAuthenticator|AbstractCoordinator|TransactionManager|AbstractStickyAssignor|DefaultSslEngineFactory|Authorizer|RecordAccumulator|MemoryRecords|FetchSessionHandler|MockAdminClient).java"/>
+              files="(AbstractFetch|ClientTelemetryReporter|ConsumerCoordinator|CommitRequestManager|FetchCollector|OffsetFetcherUtils|KafkaProducer|Sender|ConfigDef|KerberosLogin|AbstractRequest|AbstractResponse|Selector|SslFactory|SslTransportLayer|SaslClientAuthenticator|SaslClientCallbackHandler|SaslServerAuthenticator|AbstractCoordinator|TransactionManager|AbstractStickyAssignor|DefaultSslEngineFactory|Authorizer|RecordAccumulator|MemoryRecords|FetchSessionHandler|MockAdminClient).java"/>
```

Comment:
why are we suppressing this? Ideally we would like to follow the same styling as rest of the code and hence fix the checkstyle failures.

Reply:
How about removing the trace out completely? I'm not a fan of calling trace in such a tight loop

Reply:
I am not sure who might be relying on this log and I am not aware of the original motivation of adding this. I would prefer to keep the current state and only focus on perf improvement in this PR.

Reply:
If you prefer to keep the trace logic, the added code is purely to reduce memory allocation but it comes with extra baggage for additional cyclomatic complexity and hence the suppression

Reply:
Understood. Thanks. This answers my question on my we are suppressing it.